src/test/java/com/ericsson/gerrit/plugins/eiffel/listeners/TestAbstractEventListener.java

```diff
@@ -0,0 +1,179 @@
+        when(pluginConfig.isEnabled()).thenReturn(true);
+        when(pluginConfig.getFilter()).thenReturn("nope nupe my-branch nepp nupp");
+
+        enabled = listenerTestMock.verifyPluginEnabled(changeMergedEvent, pluginConfig);
+        assertTrue("Plugin should be enabled config filter and branch from GerritEvent match.",
+                enabled);
+    }
+
+    @Test
+    public void testVerifyPluginEnabledForMyBranchUsingMultipleFilterAndRegex() throws Throwable {
+        boolean enabled;
+        when(pluginConfig.isEnabled()).thenReturn(true);
+        when(pluginConfig.getFilter()).thenReturn("nope nupe (my-).* nepp nupp");
+
+        enabled = listenerTestMock.verifyPluginEnabled(changeMergedEvent, pluginConfig);
+        assertTrue("Plugin should be enabled config filter and branch from GerritEvent match.",
+                enabled);
+    }
+
+    @Test
+    public void testVerifyPluginDisableForAnotherBranch() throws Throwable {
+        boolean enabled;
+        when(pluginConfig.isEnabled()).thenReturn(true);
+        when(pluginConfig.getFilter()).thenReturn("another-branch");
+
+        enabled = listenerTestMock.verifyPluginEnabled(changeMergedEvent, pluginConfig);
+        assertFalse(
+                "Plugin should be disabled when config filter and branch from GerritEvent does not match.",
+                enabled);
+    }
+
+    @Test
+    public void testVerifyPluginEnabledForAnotherBranchNoFilter() throws Throwable {
+        boolean enabled;
+        when(pluginConfig.isEnabled()).thenReturn(true);
+        when(pluginConfig.getFilter()).thenReturn("");
+
+        enabled = listenerTestMock.verifyPluginEnabled(changeMergedEvent, pluginConfig);
+        assertTrue("Plugin should be enabled when config filter is empty.", enabled);
+    }
+
+    @SuppressWarnings("unchecked")
+    private void setUpMocks() {
+        pluginConfig = mock(EiffelPluginConfiguration.class);
+        changeMergedEvent = mock(ChangeMergedEvent.class);
+        supplierChangeAttribute = mock(Supplier.class);
+        changeAttribute = mock(ChangeAttribute.class);
+        supplierPatchSetAttribute = mock(Supplier.class);
+        patchSetAttribute = mock(PatchSetAttribute.class);
+        accountAttribute = mock(AccountAttribute.class);
+
+        when(supplierChangeAttribute.get()).thenReturn(changeAttribute);
+        when(supplierPatchSetAttribute.get()).thenReturn(patchSetAttribute);
+    }
+
+    private void populateChangeMergedEvent() {
+        changeMergedEvent.newRev = "00000000-0000-0000-0000-000000000000";
+        changeMergedEvent.change = supplierChangeAttribute;
+        changeAttribute.project = "my-project";
+        changeAttribute.branch = "my-branch";
+        changeAttribute.url = "http://my-url.com";
+        changeMergedEvent.patchSet = supplierPatchSetAttribute;
+        patchSetAttribute.author = accountAttribute;
+        accountAttribute.username = "my-user";
+        accountAttribute.email = "my@email.com";
```

Comment:
Consider making these values static final. We had them like this originally in our PR but later changed them.

Reply:
Gerritevent creation for tests should be made in seperate class not to duplicate common code. But not in this PR.